tensorboard/plugins/text_v2/text_v2_plugin_test.py

```diff
@@ -0,0 +1,230 @@
+                if include_text:
+                    summ = sess.run(
+                        vector_summary, feed_dict={placeholder: vector_message}
+                    )
+                    writer.add_summary(summ)
+
+                summ = sess.run(scalar_summary, feed_dict={placeholder: []})
+                writer.add_summary(summ)
+
+    @with_plugin()
+    def testRoutesProvided(self, plugin):
+        routes = plugin.get_plugin_apps()
+        self.assertIsInstance(routes["/tags"], collections.abc.Callable)
+        self.assertIsInstance(routes["/text"], collections.abc.Callable)
+
+    @with_plugin()
+    def testIndex(self, plugin):
+        index = plugin.index_impl(experiment="123")
+        self.assertItemsEqual(["fry", "leela"], index.keys())
+        self.assertItemsEqual(["message", "vector"], index["fry"])
+        self.assertItemsEqual(["message", "vector"], index["leela"])
+
+    @with_plugin()
+    def testText(self, plugin):
+        fry = plugin.text_impl("fry", "message", experiment="123")
+        leela = plugin.text_impl("leela", "message", experiment="123")
+        self.assertEqual(len(fry), 4)
+        self.assertEqual(len(leela), 4)
+        for i in range(4):
+            self.assertEqual(fry[i]["step"], i)
+            self.assertEqual(leela[i]["step"], i)
+
+        table = plugin.text_impl("fry", "vector", experiment="123")[0][
+            "string_array"
+        ]
+        self.assertEqual(table, [b"one", b"two", b"three", b"four"])
+
+        original_shape = plugin.text_impl("fry", "vector", experiment="123")[0][
+            "original_shape"
+        ]
+        self.assertEqual(original_shape, (4,))
+
+        truncated = plugin.text_impl("fry", "vector", experiment="123")[0][
+            "truncated"
+        ]
+        self.assertEqual(truncated, False)
+
+    def test_reduce_to_2d(self):
+        def make_range_array(dim):
+            """Produce an incrementally increasing multidimensional array.
+
+            Args:
+              dim: the number of dimensions for the array
+
+            Returns:
+              An array of increasing integer elements, with dim dimensions and size
+              two in each dimension.
+
+            Example: rangeArray(2) results in [[0,1],[2,3]].
+            """
+            return np.array(range(2 ** dim)).reshape([2] * dim)
+
+        for i in range(2, 5):
+            actual = text_v2_plugin.reduce_to_2d(make_range_array(i))
+            expected = make_range_array(2)
+            np.testing.assert_array_equal(actual, expected)
+
+    def test_reduce_and_jsonify(self):
+        convert = text_v2_plugin.reduce_and_jsonify
+        scalar = np.array("foo")
+        scalar_expected = ("foo", (), False)
+        self.assertEqual(convert(scalar), scalar_expected)
+
+        # Check that underscores are preserved correctly; this detects erroneous
+        # use of UTF-16 or UTF-32 encoding when calling markdown_to_safe_html(),
```

Comment:
We're no longer calling `markdown_to_safe_html()` so this comment (and the associated test logic) is obsolete; I would just remove this and the following three lines.

Reply:
done